File: twitter/src/main/java/com/ivansadovyi/onefeed/plugin/twitter/AccessTokenSerializer.java
package com.ivansadovyi.onefeed.plugin.twitter;

import com.migcomponents.migbase64.Base64;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

import twitter4j.auth.AccessToken;

public class AccessTokenSerializer {

	public static String serialize(AccessToken accessToken) throws IOException {
		ByteArrayOutputStream byteArrayStream = new ByteArrayOutputStream();
		ObjectOutputStream objectStream = new ObjectOutputStream(byteArrayStream);
		objectStream.writeObject(accessToken);
		byte[] accessTokenBytes = byteArrayStream.toByteArray();
		return Base64.encodeToString(accessTokenBytes, false);
	}

	public static AccessToken deserialize(String serializedData) throws IOException, ClassNotFoundException {
		byte[] accessTokenBytes = Base64.decode(serializedData);
		ByteArrayInputStream byteArrayStream = new ByteArrayInputStream(accessTokenBytes);
		ObjectInputStream objectStream = new ObjectInputStream(byteArrayStream);
		return (AccessToken) objectStream.readObject();
	}
}

File: twitter/src/main/java/com/ivansadovyi/onefeed/plugin/twitter/AccessTokenSerializer.kt
package com.ivansadovyi.onefeed.plugin.twitter

import com.migcomponents.migbase64.Base64
import twitter4j.auth.AccessToken
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import java.io.ObjectInputStream
import java.io.ObjectOutputStream

object AccessTokenSerializer {

	fun serialize(accessToken: AccessToken): String {
		val byteArrayStream = ByteArrayOutputStream()
		val objectStream = ObjectOutputStream(byteArrayStream)
		objectStream.writeObject(accessToken)
		return Base64.encodeToString(byteArrayStream.toByteArray(), false)
	}

	fun deserialize(serializedData: String): AccessToken {
		val byteArrayInputStream = ByteArrayInputStream(Base64.decode(serializedData))
		val objectStream = ObjectInputStream(byteArrayInputStream)
		return objectStream.readObject() as AccessToken
	}
}
